include/deal.II/base/symmetric_tensor.h

```diff
@@ -985,14 +996,26 @@ SymmetricTensor<rank,dim,Number>::SymmetricTensor (const Number (&array) [n_inde
 
 
 template <int rank, int dim, typename Number>
+template <typename OtherNumber>
 inline
 SymmetricTensor<rank,dim,Number> &
-SymmetricTensor<rank,dim,Number>::operator = (const Number d)
+SymmetricTensor<rank,dim,Number>::operator = (const SymmetricTensor<rank,dim,OtherNumber> &t)
 {
-  Assert (d==Number(), ExcMessage ("Only assignment with zero is allowed"));
+  data = static_cast<const base_tensor_type &>(t.data);
```

Comment:
hm, that won't work if you can't convert `Tensor<X>` into `Tensor<Y>` even if `X` can be converted to `Y`. Wouldn't you just want to do the assignment/conversion element by element?